src/main.c

```diff
@@ -201,6 +201,13 @@ shell_introspection_init (void)
 
   g_irepository_prepend_search_path (MUTTER_TYPELIB_DIR);
   g_irepository_prepend_search_path (GNOME_SHELL_PKGLIBDIR);
+
+  /* We need to explicitly add the directory where the private libraries are
+   * copied to to the GIR's library path, so that they can be found at runtime
+   * when linking using DT_RUNPATH (instead of DT_RPATH), which is the default
+   * for some linkers (e.g. gold) and in some distros (e.g. Debian).
+   */
+  g_irepository_prepend_library_path (GNOME_SHELL_PKGLIBDIR);
```

Comment:
Looks good, just prepared a PR to add that in https://github.com/endlessm/gnome-shell/pull/108